src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -48,9 +53,32 @@ public class XdocsTemplateSink extends XdocSink {
      */
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
+        this.writer = new PrintWriter(writer);
```

Comment:
@romani . Considered your suggestion. But on running the tests in my local the git diff is still occurring. Unable to find out why? Even I am forcing the `\n' EOL now

Reply:
Refreshing https://github.com/checkstyle/checkstyle/pull/13696#discussion_r1328114355
> If this is a sticking point for this PR, I recommend we restore the problem changes before executing this git diff and move this to one of the identified issues mentioned in this PR and add much more documentation of the issue and the fix needed. This issue isn't specific to this PR and has been a long problem.

Reply:
@rnveach. Fine. So, I am making another commit. Reverting all the changes related to EOL stuff and only changes related to windows script for git diff will remain in this PR.
And also, I am creating a new issue for this EOL problem.

Reply:
I would get @romani 's ok as he is running this PR review right now. I just think that we should split this apart since there is issues addressing all of this in 1 PR when this is really 2 separate issues and has been a problem for a while.

If he agrees, I would think this should just be 1 commit with the new git-diff validation, a comment above the code undoing the problem files like I mentioned, and creating or updating an existing issue with resolving the rest.

> Reverting all the changes related to EOL stuff and only changes related to windows script for git diff will remain in this PR.

Yes.

Reply:
@kalpadiptyaroy , please let me know if you need help, for now I see this PR as on your side to let you finish update.

Reply:
> @kalpadiptyaroy , please let me know if you need help, for now I see this PR as on your side to let you finish update.

https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=16642&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1768

How to get rid of this test git diff that we added to test the validation script? Apart from this. I have reverted the other changes related to EOL.